### ISSUE TITLE
Flameshot region capture shows tiny image when on dual monitor and/or screen scaling

### ISSUE THREAD

**Issue by haizaar** [flameshot-org/flameshot#2849]
### Flameshot Version

flameshot-12.1.0 (amd64)

### Installation Type

Nix

### Operating System type and version

NixOS with KDE 5.25.3 on **Wayland**

### Description

Since some recent upgrade region capture is not really useful with flameshot:

1. It shows both of my screens
2. But makes everything very small. I.e. both my 27" and 14" screen capture are displayed on half of my 27" screen.

Beforehand it only captured one of the screens but was working well in that. Now it captures both screens but it's not really useful - too tiny.

Here are my screens before capture:
![20220729_220947](https://user-images.githubusercontent.com/58201/181758137-60956721-1178-4efa-bdf7-43397b63273a.jpg)

And here is how capture UI looks:
![20220729_221310](https://user-images.githubusercontent.com/58201/181757737-93c62f3d-6b55-4494-bc00-6082f1d9822c.jpg)

My screens configuration is below:
![Screenshot_20220729_222107](https://user-images.githubusercontent.com/58201/181757233-2cc04832-e0fe-44be-b84d-60da1857c111.png)
![Screenshot_20220729_222124](https://user-images.githubusercontent.com/58201/181757278-55f6e511-a79c-4eba-bc72-e75d4e92abf4.png)

### Steps to reproduce

1. Run `flameshot`
2. Click on its systray icon

### System Information

- NixOS unstable
- KDE 5.25.3

**Comment by mmahmoudian** [comment]
Please consider:
1. provide full information (for example, you didn't mention if you are on Wayland or not)
2. write a better title for the issues. The title should contain the problem, not a general complaint.

I would be thankful if you fix these.

**Comment by haizaar** [comment]
Thanks for your feedback. I've updated the title and added "wayland" in description. Is there anything else you believe is missing?

The issue can be triggered either by 200% scale on a single/dual screen, or by just dual screen in general (even when both are on 100%). Here is the summary of my tests:
* External screen only 27" 2560x1440 @ 100% - works fine
* Laptop screen only 14" 3840x2400 @ 100% - works fine (this setup is not really usable in general)
* Laptop screen only 14" 3840x2400 @ **200**% - *broken* as per my report. The captured screen is shown on the quarter of the monitor
* Both laptop and external monitor, both at **100**% - broken as per original report. This time the captured screen is shown on the laptop monitor

**Comment by mnixry** [comment]
Same issue here, is any update on this?

**Comment by ProfessorSniff** [comment]
also encountering this issue, arch kde. scaling is at 150%, and 100% isn't practical on a laptop screen. 
![image](https://user-images.githubusercontent.com/65641177/186766780-b6ee562f-47c7-42a7-abb7-fc7f42c49ca0.png)

this issue first started happening several weeks ago, before that, it was fine. it seems like something changed, but i'm not sure if it was here or in kwin.

**Comment by Angeall** [comment]
I have the exact same issue using Windows 10 since I updated to 12.1.0

**Comment by hhsnow** [comment]
I believe I'm also seeing this issue on MacOS

MacOS 12.6
Flameshot v12.1.0 (96c2c82e) (I believe I also saw on v12.0)
2 displays - Primary 4K external monitor, secondary macbook pro screen

Full display screenshot (mac screenshot tool):
<img width="1282" alt="Screen Shot 2022-09-20 at 11 00 12" src="https://user-images.githubusercontent.com/4611809/191294474-b00d9801-e35d-4566-ad29-099a081f8cd0.png">

Flameshot selection window:
<img width="2560" alt="Screen Shot 2022-09-20 at 10 59 54" src="https://user-images.githubusercontent.com/4611809/191294704-75d77bda-a6db-4013-82d3-ae2cc04c34be.png">

Flameshot result:
![2022-09-20_11-00](https://user-images.githubusercontent.com/4611809/191294786-d2c85dd1-fb89-4c9e-8e40-8364e4613a4a.png)

**Comment by borgmanJeremy** [comment]
This is a duplicate of #564

**Comment by haizaar** [comment]
I'm not sure it's duplicate - it happens on 100% scaling as well when on dual monitor setup.

**Comment by Avimitin** [comment]
I don't think this is duplicate. Actually flameshot is working, it just shrink all the visible screen sizes and make it hard to select capture area.

**Comment by mmahmoudian** [comment]
I reopened this due to the last two comments. 

@Avimitin would you please also provide your system info and installation method and Flameshot `--version`

**Comment by Avimitin** [comment]
I am running flameshot on Arch Linux, with KDE + Wayland environment. The flameshot executable was installed via pacman, and here is the version information.

```text
Flameshot v12.1.0 (-)
Compiled with Qt 5.15.5
```

This is how it looks like when I run command `flameshot gui`:

![IMG_20230122_172720](https://user-images.githubusercontent.com/30021675/213908919-8faeedf0-e891-4c78-a668-11c5d00882c2.jpg)

Also there are some extra logging when I run the `flameshot gui` command:

```text
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
```

**Comment by Chryseus** [comment]
I'm also having a very similar problem, using Arch with KDE Wayland as well, both monitors are partly merged together and displayed on the primary display, scaling is 100% on both.

![flameshot-bug](https://user-images.githubusercontent.com/44627250/214029343-f2137d82-3536-4178-a7bc-761561c02265.jpg)

Display setup:
![image](https://user-images.githubusercontent.com/44627250/214029621-22794623-38bd-4546-a6d8-d40a0e07d6ce.png)

Also getting `qt.qpa.wayland: Wayland does not support QWindow::requestActivate()` as well.

**Comment by ThatFruedDued** [comment]
Same issue on Arch with KDE Wayland, scaling set to 125%. Running Flameshot v12.1.0. Getting same `qt.qpa.wayland: Wayland does not support QWindow::requestActivate()` as others here. Problem did not occur for me on 100% scaling, only started appearing when I switched to 125%.

![2023-03-14_22-42](https://user-images.githubusercontent.com/57722939/225191641-a65cf33f-af60-48fa-9b3d-760e712dc7fb.png)
![2023-03-14_22-40](https://user-images.githubusercontent.com/57722939/225191648-f9e113f6-b7a8-470d-baa6-11fd1bf7efa1.png)
![Screenshot_20230314_224020](https://user-images.githubusercontent.com/57722939/225191649-96e77269-6199-4fef-a435-392971823318.png)

**Comment by mmahmoudian** [comment]
> Problem did not occur for me on 100% scaling, only started appearing when I switched to 125%.

That is an issue with fractional scaling. Is 200% also problematic?

**Comment by ThatFruedDued** [comment]
> That is an issue with fractional scaling. Is 200% also problematic?
Yes, 200% scaling has the same issue, where the image appears in the top-left corner. 100% works fine.

**Comment by shitan198u** [comment]
im also facing the same issue on any scaling the capture window becomes tiny it doesnt happen in 100% scaling

**Comment by sung-kim** [comment]
+1 I'm facing the same issue on MacOS 14.2.1, info from flameshot "About" menu: 

```
Flameshot v12.1.0 (96c2c82e)
Compiled with Qt 5.15.5
darwin: 23.2.0
osx: 14.2
```

**Comment by HEIlZMaker** [comment]
This bug occurs with a single monitor at scale 125% also

I am running debian 12 KDE

installed via "sudo nala install flameshot"
The output of flameshot -v is:
Flameshot v12.1.0 (Debian 12.1.0-2)
Compiled with Qt 5.15.8

Here
![flameshotbug](https://github.com/flameshot-org/flameshot/assets/78142078/5322b950-02d5-4684-9f3d-775386c1dbfa)

**Comment by tirtharajsinha** [comment]
This issue is still there and have not fixed.

**Comment by floshipper** [comment]
Same issue on macos remains

**Comment by SwapnilSoni1999** [comment]
Same issue still remains even on single screen with wayland on arch with KDE plasma 6 that too with a single screen (I only have my laptop)

**Comment by hwk2077** [comment]
same and still unfixed

**Comment by fndribeiro** [comment]
same

**Comment by ShadowPlagueXx** [comment]
still have the same issue here

**Comment by stevesimmons** [comment]
Is this regarded as a live issue to be fixed?
I get it using Windows 11 with laptop screen plus a large external monitor at 250% / 150% scaling.
Unfortunately, it makes FlameShot unusable for me when connected to the external monitor.

**Comment by domeniczz** [comment]
PLEASE fix the issue, encountering the exact same issue here

**Comment by Umbyr93** [comment]
I guess this isn't fixed yet? I'm having the same problem.

**Comment by Charles-h1r0** [comment]
same

**Comment by paulohenriquepm** [comment]
same here

**Comment by AJR07** [comment]
^^^^ Same issue, please fix it

**Comment by phasephasephase** [comment]
same issue on my laptop's integrated screen

**Comment by moalhaddar** [comment]
Can confirm that this is happening to me. 

Happens with a single monitor and scaling set to 125% on Gnome 47.

The issue stops from happening with scaling set to 100%.

**Comment by TackyTortoise** [comment]
I'm encountering the same issue on EndeavorOS using KDE Plasma at 115% scale

**Comment by f1sherb0y** [comment]
same issue on Arch Linux using KDE plasma at 150% scale

![Image](https://github.com/user-attachments/assets/509624a0-8421-4d86-9c77-01e9c1e86f59)

**Comment by mateusfg7** [comment]
Same here :(

**125% Scaling**, works if set 100% scaling, but I can't use my laptop on 100% scaling, it gets too small.

![Image](https://github.com/user-attachments/assets/85e408a8-bc96-412e-ba60-f81525f4f8f7) ![Image](https://github.com/user-attachments/assets/28335703-cf63-4b3c-945d-f68353698631)

**Comment by isaevdan** [comment]
Have same issue. In my case I have laptop connected to monitor (monitor uses scaling) and I use display setting to show only monitor when connected. So I can start my laptop with flameshot running and then connect monitor (maybe works with initially connected monitor as well), after this I have clipped area. My guess is that flameshot doesn't update scaling/resolution if it was change after application start.

I use workaround to just close and open application and then on new start it picks up current display settings and everything works fins. Maybe this will be useful to someone

**Comment by SwapnilSoni1999** [comment]
I guess it's still not fixed yet

**Comment by pCoder3000** [comment]
same issue here 175 scale fedora gnome

**Comment by mmahmoudian** [comment]
We have merged a fix for fractional scaling (#3869). Please test the nightly build (or AUR if you are on an Arch based distro) and let us know if the issue persists.

**Comment by gehoernchen** [comment]
I just installed flameshot-git via AUR and it looks even worse than before.

With the flameshot package, it looks exactly like the OP's picture, but with flameshot-git, part of the first monitor is used as capturing area and the other monitor is not used at all.

Like many before me, I am on
- 1080p at 100 % scaling
- 1440p at 125 % scaling
- Wayland, KDE Plasma 6.3.5

Though, even at 100 % scaling, it looks the same way.

![Image](https://github.com/user-attachments/assets/d7b6aabf-4766-4eb7-abc4-6216f116b3f3)

**Comment by mmahmoudian** [comment]
@gehoernchen 
This has nothing to do with fractional scaling. Did you by any chance connected the right monitor to your computer **after** launching Flameshot daemon?

Also can you provide the Flameshot version. Because the current AUR should not build properly as I haven't found the time to update it to use Qt6

**Comment by gehoernchen** [comment]
@mmahmoudian 
Thanks for answer - I experience the same problem with fractional scaling as this issue reported, but had hoped your suggestion fixes it, which makes it worse.
No, I connected it before turning on the system. Main monitor (left) via DP, right one via HDMI.

```
$ yay -Qm flameshot-git
flameshot-git r2057.58afdce7-1

$ flameshot -v     
Flameshot v12.1.0 (58afdce7)
Compiled with Qt 6.9.1
```

**Comment by iongion** [comment]
Me too - my screen goes split into 4, previous release did not have it, I have dual monitor

<img width="880" height="378" alt="Image" src="https://github.com/user-attachments/assets/f05133d7-8970-4b7e-ae94-afd7af33f831" />

<img width="3839" height="2159" alt="Image" src="https://github.com/user-attachments/assets/9fc93765-0bf0-4d6e-b8b9-2dc7a867ac0f" />

I now have Flameshot v13.1.0 (79395fd3) obtained with winget, I have no idea what was the last version I had, but that one did not have the bug. Up pressing Take Screenshot, it is rescaling the entire desktop to this weird 4 areas size.

**Comment by M1ndo** [comment]
This issue still persists, scaling or without scaling.
My screen get smaller, 

Running on Arch Linux, Xmonad window Manager at (3840x2400@60), no QT Scaling applied, even with Scaling same issue.

<img width="3840" height="2400" alt="Image" src="https://github.com/user-attachments/assets/d71dd797-b1fc-4f64-98ca-bf277657e805" />

**Comment by DryPringleSoup** [comment]
flameshot-13.1.0-1, this has become an issue. I haven't changed any screen settings but this just started happening a couple days ago.

I'm on Arch Linux with openbox. My screen settings are in the first image: 3840x2400, 60.00 Hz, 1.00 scaling

<img width="868" height="539" alt="Image" src="https://github.com/user-attachments/assets/5a076cb3-bc88-4f44-9cc8-33d50241e826" />

<img width="3197" height="2150" alt="Image" src="https://github.com/user-attachments/assets/d5347541-c0f2-41a2-afaa-d74f01b5771d" />

**Comment by M1ndo** [comment]
@DryPringleSoup I have come around this by using `maim` to take screenshots, this works great just without the fancy flameshot things.

You can find scripts `dmenu-scrot.sh` to take either a full screenshot/region ... make sure u also get `config` 

https://github.com/M1ndo/dotfiles/tree/master/.dmenu

hope it helps :)

**Comment by hownioni** [comment]
I can report the same issue on Arch Linux using X11, QTile and flameshot v13.1.0. I'm using 2.00 scaling

**Comment by chalofa** [comment]
same bug, using Arch Linux, 1.25 scaling and X11 and AUR flameshot package:

```
❯ flameshot -v
Flameshot v13.1.0 (eb4ed57)
Compiled with Qt 6.9.1
```

issue persist even if I disconnect external monitor, and I can confirm it's related to fractional scaling... weird thing is taht this didn't happen to me 2 weeks ago, so a recent update started affecting me

**Comment by hownioni** [comment]
> same bug, using Arch Linux, 1.25 scaling and X11 and AUR flameshot package:
> 
> ```
> ❯ flameshot -v
> Flameshot v13.1.0 (eb4ed57)
> Compiled with Qt 6.9.1
> ```
> 
> issue persist even if I disconnect external monitor, and I can confirm it's related to fractional scaling... weird thing is taht this didn't happen to me 2 weeks ago, so a recent update started affecting me

Same for me here. The bug started not too long ago. Before that, it was fine

**Comment by DryPringleSoup** [comment]
@M1ndo That does seem like a nice alternative, but the entire reason I use Flameshot instead of anything else is _because_ it has all those handy features (like quick annotation, pinning, and others) which have kept me from switching to a leaner program.

**Comment by likle** [comment]
This fixes it for me:

`env QT_ENABLE_HIGHDPI_SCALING=0 flameshot gui`

**Comment by hownioni** [comment]
> This fixes it for me:
> 
> `env QT_ENABLE_HIGHDPI_SCALING=0 flameshot gui`

Thanks! This worked for me too :3

**Comment by brunolmadeira** [comment]
> This fixes it for me:
> 
> `env QT_ENABLE_HIGHDPI_SCALING=0 flameshot gui`

Worked here too. Thanks!

OS: CachyOS
DE: GNOME
WM: x11
Scaling: 1.25

**Comment by DryPringleSoup** [comment]
Ok, it's been a couple weeks and the bug hasn't been fixed. I've tried @likle solution to no avail so the only working fix, for me, is reverting flameshot to a previous version. The lastest version that doesn't seem to have this bug is v13.0.1-1
```
sudo pacman -S downgrade
sudo downgrade 'flameshot=13.0.1-1' --latest --prefer-cache --ignore always
```

**Comment by chalofa** [comment]
> ```
> sudo downgrade 'flameshot=13.0.1-1' --latest --prefer-cache --ignore always
> ```

thanks! this finally fixed my issue!

**Comment by iongion** [comment]
Nothing works properly still

- **Option 1**: `QT_ENABLE_HIGHDPI_SCALING=0` does not really work - although the scaling issue appears do be gone, the actual cropped area is not the proper resolution
- **Option 2**: Downgrading to `13.0.1` also does not work, while the scaling issue is gone too, on my system with 2 monitors, when I try to capture, the window content is moved between monitors, flame-shot appears confused. This is the closest to how it should work, but the monitor swap is not good

**Comment by DryPringleSoup** [comment]
Also could someone remove the Wayland label on this issue? I got the bug on an X11 system.

**Comment by hownioni** [comment]
> Nothing works properly still
> 
>     * **Option 1**: `QT_ENABLE_HIGHDPI_SCALING=0` does not really work - although the scaling issue appears do be gone, the actual cropped area is not the proper resolution
> 
>     * **Option 2**: Downgrading to `13.0.1` also does not work, while the scaling issue is gone too, on my system with 2 monitors, when I try to capture, the window content is moved between monitors, flame-shot appears confused. This is the closest to how it should work, but the monitor swap is not good

What do you mean the area cropped is not the proper resolution in Option 1? For me at least it is

**Comment by iongion** [comment]
@hownioni It is not in the sense that if I capture 4K images, the resolution of the captured cropped region is not respecting the 4K image, it is downscaled

**Comment by ashayh** [comment]
Same on KDE plasma Ubuntu 25.10 with 125% scaling on Nvidia with a 4k monitor.
```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 25.10
Release:        25.10
Codename:       questing

flameshot -v 
Flameshot v12.1.0 (Debian 12.1.0+ds-2)
Compiled with Qt 5.15.15


```

Changing scale to 100% fixes this issue.

**Comment by mmahmoudian** [comment]
This should have been fixed with #4498 
If you can, try compiling from the master branch and let us know is this issue still persists. I'm going to close this for now, but if the issue persists, we will re-open it.
